### ISSUE TITLE
same issue with #663 ,root can not be a array

### ISSUE THREAD

**Issue by DusuWen** [josdejong/jsoneditor#687]
![image](https://user-images.githubusercontent.com/15500093/55770743-0708a480-5ab8-11e9-9f4f-4d11230e1199.png)
![image](https://user-images.githubusercontent.com/15500093/55770758-112aa300-5ab8-11e9-8970-aeb811b88b9e.png)
![image](https://user-images.githubusercontent.com/15500093/55771259-e04b6d80-5ab9-11e9-9447-7a9d472b7319.png)

i use scheme to validate input json,but thres is error in console,here is my code
```json
{
  "definitions": {},
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$id": "http://example.com/root.json",
  "type": "array",
  "title": "The Root Schema",
  "items": {
    "$id": "#/items",
    "type": "object",
    "title": "The Items Schema",
    "required": [
      "name"
    ],
    "properties": {
      "name": {
        "$id": "#/items/properties/name",
        "type": "string",
        "title": "The Name Schema",
        "default": "",
        "examples": [
          "one"
        ],
        "pattern": "^(.*)$"
      }
    }
  }
}
```
here is input json 
```json
[
  {
    "name": "one"
  },
  {
    "name": "two"
  }
]
```

**Comment by josdejong** [comment]
Thanks for reporting @DusuWen. Just to be sure: are you using the latest version of JSONEditor? Similar Invalid JSON path errors where solved in one of the latest versions.

**Comment by DusuWen** [comment]
i m sure is 5.32.1@josdejong

**Comment by josdejong** [comment]
Thanks for checking. I think this issue is the same as #682: when the schema throws an error because of invalid data, JSONEditor fails to find the node where to attach this error.

**Comment by AnthonyQ96** [comment]
wow, i have same question,  Do you cope? @DusuWen

**Comment by DusuWen** [comment]
hi @josdejong ,i check out source code,seems like function parsePath can not parse `jsonPath = "[0]"`,am i right?
![image](https://user-images.githubusercontent.com/15500093/55865870-cd1bc900-5bb1-11e9-9c4c-b5aee9f66437.png)
![image](https://user-images.githubusercontent.com/15500093/55866537-220c0f00-5bb3-11e9-8a26-d0b84367dc62.png)

**Comment by DusuWen** [comment]
i deleted  `i > 0 &&` form `i > 0 && jsonPath[i] === '['` then it works as expected. @josdejong 
![image](https://user-images.githubusercontent.com/15500093/55869070-5930ef00-5bb8-11e9-96a6-e3f78c13a64a.png)

**Comment by josdejong** [comment]
Ahh, that's it @DusuWen , thanks for debugging this issue! I've worked out your fix and added unit tests in this commit: 70b7e7914b7f2f2453cbaa475e50d9c1dc11ebf0.

**Comment by josdejong** [comment]
Should be fixed now in `v5.32.4`, thanks again @DusuWen

**Comment by DusuWen** [comment]
Thanks ~~ @josdejong
